core/src/main/java/org/elasticsearch/action/support/replication/ReplicationOperation.java

```diff
@@ -284,6 +283,11 @@ private void decPendingAndFinishIfNeeded() {
 
     private void finish() {
         if (finished.compareAndSet(false, true)) {
+            try {
+                primary.updateGlobalCheckpoint();
```

Comment:
@ywelsch I discussed this with @bleskes. We are going to keep this as-is for now. If we see that the lag until the shard goes idle is a problem, we can address it then.